### DISCUSSION TITLE
Streaming - finish processing whatever data is still has cached without calling CloseStream

### DISCUSSION THREAD

**Discussion by cheng-talkstack** [deepgram/community#487] [General help] [answered]
From your documentation

> When you are finished, send a JSON message to the server: { "type": "CloseStream" }. The server will interpret it as a shutdown command, which means it will finish processing whatever data is still has cached, send the response to the client, send a summary metadata object, and then terminate the WebSocket connection.

Is it possible to clear the cache without closing the WebSocket connection?

My problem is that restarting WebSocket connection to Deepgram take a considerable time for it to start processing again.

I am using Twilio that send a mark message, when I received this mark message back from Twilio, and start WebSocket connection to Deepgram again, it take a bit of lag before it start to recognize speech again.

My solution was to keep Deegram WebSocket connection alive but the transcript from before the mark message send from Twilio will also be inside the result from Deepgram, so I want a way to clear the transcript cache without closing the WebSocket connection.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by nikolawhallon** [reply, marked as answer]
I agree that reconnecting is a pain and introduces latency, and this feature has been requested a few times, I think it would be an awesome feature and I think we will work on implementing it, but currently there are no promises in terms of a timeline, we will update when we are able. Great feature suggestion.

**Reply by cheng-talkstack** [reply]
Hello @nikolawhallon thank you for considering this feature request, it would be great if we can send { "type": "ClearCache" } message to Deepgram WebSocket. I hope to hear good news from the team soon.
